exchanges/orderbook/depth.go

```diff
@@ -784,11 +786,29 @@ func (d *Depth) GetTranches(count int) (ask, bid []Tranche, err error) {
 }
 
 // GetPair returns the pair associated with the depth
-func (d *Depth) GetPair() (currency.Pair, error) {
+func (d *Depth) GetPair() currency.Pair {
 	d.m.RLock()
 	defer d.m.RUnlock()
-	if d.pair.IsEmpty() {
-		return currency.Pair{}, currency.ErrCurrencyPairEmpty
-	}
-	return d.pair, nil
+	return d.pair
+}
+
+// GetAsset returns the asset associated with the depth
+func (d *Depth) GetAsset() asset.Item {
+	d.m.RLock()
+	defer d.m.RUnlock()
+	return d.asset
+}
+
+// GetExchange returns the exchange associated with the depth
+func (d *Depth) GetExchange() string {
+	d.m.RLock()
+	defer d.m.RUnlock()
+	return d.exchange
+}
+
+// GetKey returns the key associated with the depth
+func (d *Depth) GetKey() key.ExchangePairAsset {
```

Comment:
Note rewording comment, since there is no key associated, it felt misleading.
```suggestion
// Key returns a combined key for the depth
func (d *Depth) Key() key.ExchangePairAsset {
```

Reply:
[done](https://github.com/thrasher-corp/gocryptotrader/pull/1783/commits/bddca75587681ceee868a1e9dc5dc4d817a6e039)